src/components/MapView.tsx

```diff
@@ -98,24 +98,17 @@ export const MapView = () => {
     setActiveStop(null);
   };
 
-  const createConnection = (
-    coordinates: {lat: number; lon: number},
-    id: string,
-    stopType: number,
-    name: string,
-    ref: string,
-  ) => {
+  const createConnection = (busStop: Stop) => {
     if (connectionData.length < 2) {
-      const isOsm = stopType === 0;
-      const entryPoint = {coordinates, id, isOsm, name, ref};
-
-      if (connectionData.length === 1 && connectionData[0].isOsm === isOsm) {
-        if (connectionData[0].id !== id) {
+      // Check if stop is the same type as selected one
```

Comment:
Do we really need these comments? They just explain what code does, nothing more. If code is too ambiguous, how about making one-liner function like
`function sameStopClicked (connectionData) {return connectionData.length === 1 && connectionData[0].stopType === busStop.stopType) }`
Then: `if (sameStopClicked(connectionData)) { ... }`

Reply:
It's good suggestions but I prefer these comments - it's shorter just in this case.